app/controllers/payment-links/metadata/metadata-form.js

```diff
@@ -1,19 +1,42 @@
+const META_DATA_MAX_LENGTH = 50
+const MAX_NO_OF_METADATA_COLUMNS = 10
+
 const fields = {
   metadataKey: {
     id: 'metadata-column-header',
     name: 'metadata-column-header',
-    validation: [ {
-      validator: isNotEmpty,
-      message: 'Enter a column header'
-    } ]
+    validation: [
+      {
+        validator: isNotEmpty,
+        message: 'Enter a column header'
+      },
+      {
+        validator: isValidLength,
+        message: `Column header must be ${META_DATA_MAX_LENGTH} characters or fewer`
+      },
+      {
+        validator: isNotDuplicate,
+        message: 'Column header must not already exist'
+      },
+      {
+        validator: isNotExceedingMaxNoOfReportingColumns,
```

Comment:
As per feedback,  I have added an extra validator - to check that the number of reporting columns does not exceed 10.